nomad/server.go

```diff
@@ -271,6 +279,9 @@ func NewServer(config *Config, consulCatalog consul.CatalogAPI, logger *log.Logg
 	// Create the periodic dispatcher for launching periodic jobs.
 	s.periodicDispatcher = NewPeriodicDispatch(s.logger, s)
 
+	// Initialize the stats fetcher that autopilot will use.
+	s.statsFetcher = NewStatsFetcher(logger, s.connPool, s.config.Region)
+
 	// Setup Vault
 	if err := s.setupVaultClient(); err != nil {
```

Comment:
The `Leave` method here also calls RemovePeer and currently does not handle raft protocol 3, I left a TODO in there in the master branch.

Reply:
Another TODO https://github.com/hashicorp/nomad/blob/master/api/operator.go#L72

Reply:
The last of these (raft ID related) should be taken care of now